File: src/game/tools/rpc/output/wrap/ArenaWrap.cs
using System;
using Common.Net.Simple;

namespace Game 
{
	public class ArenaWrap : IWrapper
	{	
		public BattleHeroWrap[] Heroes; // 可选英雄
		public BattleHeroWrap[] OnBattleHeroes; // 上阵英雄
		
		public void Decode(Packet pck)
		{	
			Heroes = new BattleHeroWrap[pck.GetShort()];
			for (int i = 0; i < Heroes.Length; i++)
			{
				Heroes[i] = new BattleHeroWrap();
				Heroes[i].Decode(pck);
				
			}
			
			OnBattleHeroes = new BattleHeroWrap[pck.GetShort()];
			for (int i = 0; i < OnBattleHeroes.Length; i++)
			{
				OnBattleHeroes[i] = new BattleHeroWrap();
				OnBattleHeroes[i].Decode(pck);
				
			}
			
		}
		
        public void Encode(Packet pck)
        {	
        	if (Heroes == null) pck.PutShort((short)0); 
        	else
        	{
	        	pck.PutShort((short)Heroes.Length);
	        	for(int i = 0; i < Heroes.Length; i++)
	        	{
	        		Heroes[i].Encode(pck);
					
	        	}
	        }
        	
        	if (OnBattleHeroes == null) pck.PutShort((short)0); 
        	else
        	{
	        	pck.PutShort((short)OnBattleHeroes.Length);
	        	for(int i = 0; i < OnBattleHeroes.Length; i++)
	        	{
	        		OnBattleHeroes[i].Encode(pck);
					
	        	}
	        }
        	
        }
	}
}

File: src/game/tools/rpc/output/wrap/ArenaWrap.go
package wraps

import (
	rpc "tipcat.com/common/rpc/simple"
)

type ArenaWrap struct {
	
	Heroes []*BattleHeroWrap // 可选英雄
	OnBattleHeroes []*BattleHeroWrap // 上阵英雄
	rpc.Wrapper
}

func (w *ArenaWrap)Decode(pck *rpc.Packet) rpc.Wrapper {
	
	w.Heroes = make([]*BattleHeroWrap, int(pck.PopInt16()))
	for i := 0; i < len(w.Heroes); i++ {
		w.Heroes[i] = new(BattleHeroWrap)
		w.Heroes[i].Decode(pck)
		
	}
	
	w.OnBattleHeroes = make([]*BattleHeroWrap, int(pck.PopInt16()))
	for i := 0; i < len(w.OnBattleHeroes); i++ {
		w.OnBattleHeroes[i] = new(BattleHeroWrap)
		w.OnBattleHeroes[i].Decode(pck)
		
	}
	
	return w
}

func (w *ArenaWrap)Encode(pck *rpc.Packet) {
	
	if w.Heroes == nil {
		pck.PutInt16(0)
	} else {
    	pck.PutInt16(int16(len(w.Heroes)))
    	for i := 0; i < len(w.Heroes); i++ {
    		w.Heroes[i].Encode(pck);
			
    	}
    }
	
	if w.OnBattleHeroes == nil {
		pck.PutInt16(0)
	} else {
    	pck.PutInt16(int16(len(w.OnBattleHeroes)))
    	for i := 0; i < len(w.OnBattleHeroes); i++ {
    		w.OnBattleHeroes[i].Encode(pck);
			
    	}
    }
	
}

